src/util-ja3.c

```diff
@@ -300,3 +302,34 @@ InspectionBuffer *Ja3DetectGetString(DetectEngineThreadCtx *det_ctx,
     }
     return buffer;
 }
+
+#else /* HAVE_JA3 */
+
+/* Stubs for when JA3 is disabled */
+
+int Ja3BufferAppendBuffer(JA3Buffer **buffer1, JA3Buffer **buffer2)
+{
+    return 0;
+}
+
+int Ja3BufferAddValue(JA3Buffer **buffer, uint32_t value)
+{
+    return 0;
+}
+
+char *Ja3GenerateHash(JA3Buffer *buffer)
```

Comment:
Why do we need this ifndef HAVE_JA3 ?

Reply:
I wanted to keep the behaviour as close as possible to the `HAVE_JA3` case. But I guess we can also pass `NULL` here, given that it won't be accessed in that case and also will only be freed if it is non-`NULL`.

Reply:
But why should we need an empty `Ja3GenerateHash` if ja3 is not enabled ?

Reply:
See above - I didn't want to have `#ifdefs` in the parser code and just base the behaviour on the implementation of these functions.